src/main/webapp/app/iris/manage/settings/shared/iris-enabled/iris-enabled.component.ts

```diff
@@ -7,60 +7,123 @@ import { IrisSettingsService } from 'app/iris/manage/settings/shared/iris-settin
-    @Input() disabled? = false;
+    exercise = input<Exercise | undefined>();
+    course = input<Course | undefined>();
+    lecture = input<Lecture | undefined>();
+    irisSubSettingsType = input.required<IrisSubSettingsType>();
+    disabled = input<boolean>(false);
+    showCustomButton = input<boolean>(false);
 
     irisSettings?: IrisSettings;
     irisSubSettings?: IrisSubSettings;
+    someButNotAllSettingsEnabled = false;
 
     ngOnInit(): void {
-        if (this.exercise) {
-            this.irisSettingsService.getUncombinedExerciseSettings(this.exercise.id!).subscribe((settings) => {
+        if (this.exercise()) {
+            this.irisSettingsService.getUncombinedExerciseSettings(this.exercise()!.id!).subscribe((settings) => {
                 this.irisSettings = settings;
                 this.setSubSettings();
             });
-        } else if (this.course) {
-            this.irisSettingsService.getUncombinedCourseSettings(this.course.id!).subscribe((settings) => {
+        } else if (this.course()) {
+            this.irisSettingsService.getUncombinedCourseSettings(this.course()!.id!).subscribe((settings) => {
                 this.irisSettings = settings;
                 this.setSubSettings();
             });
         }
     }
 
     setEnabled(enabled: boolean) {
-        if (!this.disabled && this.irisSubSettings) {
+        if (!this.disabled() && this.irisSubSettings && this.irisSubSettingsType() != IrisSubSettingsType.ALL) {
             this.irisSubSettings.enabled = enabled;
-            if (this.exercise) {
-                this.irisSettingsService.setExerciseSettings(this.exercise.id!, this.irisSettings!).subscribe((response) => {
+            if (this.exercise()) {
+                this.irisSettingsService.setExerciseSettings(this.exercise()!.id!, this.irisSettings!).subscribe((response) => {
                     this.irisSettings = response.body ?? this.irisSettings;
                     this.setSubSettings();
                 });
-            } else if (this.course) {
-                this.irisSettingsService.setCourseSettings(this.course.id!, this.irisSettings!).subscribe((response) => {
+            } else if (this.course()) {
+                this.irisSettingsService.setCourseSettings(this.course()!.id!, this.irisSettings!).subscribe((response) => {
                     this.irisSettings = response.body ?? this.irisSettings;
                     this.setSubSettings();
                 });
             }
+        } else if (!this.disabled() && this.irisSubSettingsType() == IrisSubSettingsType.ALL && this.irisSettings && this.course()) {
+            // If the subsettings type is ALL, we need to set all subsettings to the same value
+            this.toggleEnabledStateForAllSubSettings(enabled);
         }
     }
 
+    private toggleEnabledStateForAllSubSettings(enabled: boolean) {
+        if (!this.irisSettings) {
+            return;
+        }
+        if (!this.irisSettings.irisChatSettings) {
```

Comment:
Did you check if this case is at all possible? Technically, the server should return default settings for all types.